src/TeamSpeak3QueryApi/Specialized/Responses/GetClientDetailedInfo.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TeamSpeak3QueryApi.Net.Specialized.Responses
+{
+    public class GetClientDetailedInfo : Response
+    {
+        [QuerySerialize("cid")]
+        public int ChannelId;
+
+        [QuerySerialize("client_unique_identifier")]
+        public string UniqueIdentifier;
+
+        [QuerySerialize("client_nickname")]
+        public string NickName;
+
+        [QuerySerialize("client_login_name")]
+        public string ServerQueryName;
+
+        [QuerySerialize("client_database_id")]
+        public int DatabaseId;
+
+        [QuerySerialize("client_type")]
+        public ClientType Type;
+
+        [QuerySerialize("client_version")]
+        public string Version;
+
+        [QuerySerialize("connection_client_ip")]
+        public string ConnectionIp;
+
+        [QuerySerialize("client_platform")]
+        public string Plattform;
+
+        [QuerySerialize("client_description")]
+        public string Description;
+
+        [QuerySerialize("client_input_muted")]
+        public bool InputMuted;
+
+        [QuerySerialize("client_output_muted")]
+        public bool OutputMuted;
+
+        [QuerySerialize("client_outputonly_muted")]
+        public bool OutputOnlyMuted;
+
+        [QuerySerialize("client_is_recording")]
+        public bool IsRecording;
+
+        [QuerySerialize("client_servergroups")]
+        public IReadOnlyList<int> ServerGroupIds;
+
+        [QuerySerialize("client_channel_group_id")]
+        public IReadOnlyList<int> ChannelGroupsIds;
+
+        [QuerySerialize("client_created")]
+        public DateTime Created;
+        
+        [QuerySerialize("client_lastconnected")]
+        public DateTime LastConnected;
+        
+        [QuerySerialize("client_totalconnections")]
+        public int TotalConnectionCount;
+        
+        [QuerySerialize("client_away")]
+        public bool Away;
+        
+        [QuerySerialize("client_away_message")]
+        public string AwayMessage;
+        
+        [QuerySerialize("connection_connected_time")]
+        private long _connectionTime; //Because it is in ms instead if s defined in Typecaster
+        public TimeSpan ConnectionTime => TimeSpan.FromMilliseconds(_connectionTime);
+
+        [QuerySerialize("client_idle_time")]
+        private long _idleTime; //Because it is in ms instead if s defined in Typecaster
+        public TimeSpan IdleTime => TimeSpan.FromMilliseconds(_idleTime);
```

Comment:
Does the API always return milliseconds? It might be possible that I haven't testes the results of `TimeSpanTypeCaster`. If so, we can just fix it in the type caster.

Reply:
I only tested it for clientinfo. Here they use secounds for the DateTimes and ms for the TimeSpans

Reply:
![image](https://user-images.githubusercontent.com/13467300/40263433-73185fe0-5b12-11e8-86a9-dfcacb6366a4.png)
it seems right for getServers

Reply:
It is weired that tey use different units for timespans
![image](https://user-images.githubusercontent.com/13467300/40263448-c7267f72-5b12-11e8-9012-89003496eeb8.png)

Reply:
The screenshots were taken with ms for Clientinfo and seconds for GetServers

Reply:
Teamspeak is reeaally inconsistent with their timespan fields. If you need any references we maintain a data file with a lot of message types and fields:
https://github.com/ReSpeak/tsdeclarations/blob/master/Messages.toml#L299
Maybe this helps clear up a bit confusion.
(We also have a template file to generate messages from this declaration file e.g: https://github.com/Splamy/TS3AudioBot/blob/master/TS3Client/Generated/Messages.cs)
We want to help all teamspeak/query related projects to set up tedious task faster; feel free to contribute if you think this would help.

Reply:
It's not the first time that I stumble upon such inconsistencies. Thanks for providing a link to your work!

Reply:
What about an Interface and than use something like this:`[QuerySerialize<FromMiliseconds>("client_idle_time")]` or propably in a subclass to improve readability: ? `[QuerySerialize<TimeConverter.FromMiliseconds>("client_idle_time")]`